smart-contracts/smart-contract/tests/main-contract_test.ts

```diff
@@ -48,6 +51,134 @@ const err_not_proposed_notifier = '(err u124)';
 const err_already_notifier = '(err u125)';
 const err_no_voting_period = '(err u129)';
 
+Clarinet.test({
+  name: 'Get All Data Waiting 300 Miners',
+  async fn(chain: Chain, accounts: Map<string, Account>) {
+    const deployer = accounts.get('deployer')!;
+    let waiting_list = [];
+    let block = chain.mineBlock([]);
+    assertEquals(block.receipts.length, 0);
+    assertEquals(block.height, 2);
+
+    for (let i = 1; i <= 299; i++) {
+      const miner = accounts.get(`wallet_${i}`)!;
+      block = chain.mineBlock([
+        Tx.contractCall(CONTRACT_NAME, ASK_TO_JOIN, [types.ascii(miner.address)], miner.address),
+      ]);
+      if (i == 1) waiting_list.push(`${miner.address}`);
+      else waiting_list.push(` ${miner.address}`);
+    }
+    block = chain.mineBlock([Tx.contractCall(CONTRACT_NAME, GET_WAITING_LIST, [], deployer.address)]);
```

Comment:
@BowTiedJohn can we add another assertEquals to check the length of the waiting list variable?